src/wms/src/test/java/org/geoserver/wms/map/RenderedImageMapOutputFormatTest.java

```diff
@@ -350,6 +350,154 @@ public void testInterpolations() throws IOException, IllegalFilterException, Exc
                 getPixelColor(imageBicubic, 300, 300).getRGB());
     }
 
+    @Test
+    public void testInterpolationFromLayerConfig() throws IOException, IllegalFilterException, Exception {
+        final Catalog catalog = getCatalog();
+
+        LayerInfo layerInfo = catalog.getLayerByName(MockData.TASMANIA_DEM.getLocalPart());
+
+        MapLayerInfo mapLayer = new MapLayerInfo(layerInfo);
+        assertNull(layerInfo.getDefaultInterpolationMethod());
+
+        Envelope env = layerInfo.getResource().boundingBox();
+        double shift = env.getWidth() / 6;
+
+        env = new Envelope(env.getMinX() - shift, env.getMaxX() + shift, env.getMinY() - shift,
+                env.getMaxY() + shift);
+
+        // set Nearest Neighbor interpolation on layer
+        GetMapRequest request = new GetMapRequest();
+        request.setFormat(getMapFormat());
+        request.setLayers(Arrays.asList(mapLayer));
+
+        layerInfo.setDefaultInterpolationMethod("Nearest");
+        assertEquals("Nearest",
+                request.getLayers().get(0).getLayerInfo().getDefaultInterpolationMethod());
+        assertTrue(request.getInterpolations().isEmpty());
+
+        WMSMapContent map = createWMSMap(env);
+        map.setRequest(request);
+
+        RenderedImageMap imageMap = this.rasterMapProducer.produceMap(map);
```

Comment:
You mean something along these lines?
```
Layer mLayer = imageMap.getMapContext().layers().get(0);
Interpolation interpolationSet = (Interpolation)mLayer.getUserData().get(StreamingRenderer.BYLAYER_INTERPOLATION);
assertNotNull(interpolationSet);
assertEquals(Interpolation.getInstance(Interpolation.INTERP_NEAREST), interpolationSet);
```

I would have to disable the direct raster render path (active by default), otherwise nothing is set on the layer... is my understanding correct?